src/client/mod.rs

```diff
@@ -30,7 +30,7 @@ use version::HttpVersion;
 
 pub use http::response::Response;
 pub use http::request::Request;
-pub use self::connect::{HttpConnector, Connect};
+pub use self::connect::{HttpConnector, Connect, HttpConnectorBlockingTask};
```

Comment:
Is it possible to not expose this type to outside the crate?

Reply:
I'm not 100% sure what you are saying here.

Not exposing it as in not exporting it, but with the type still being part of the signature of `HttpConnector::new_with_executor`? Or instead boxing so that `HttpConnector::new_with_executor` requires an `Executor<Box<Future<Item=(), Error=()>>>`?

Both don't seem ideal to me :/

Reply:
Oh OK, it's used for the generic type on `futures::future::Executor<T>`... Could the constructor `new_with_executor` just be generic over `F: Future<Item=(), Error=()>`? After that, it's turned into a trait object and put in an `Arc`, so the generic isn't really important...

Reply:
Hmm, maybe I'm misunderstanding, but as far as I can see when implementing a function that is generic over `F`, it's impossible for us to produce a future of type `F` without being passed one from the outside. But since our goal is to be able to execute arbitrary futures, I'm unsure how that would help us?

Reply:
I checked out the branch and saw what you mean. Ideally we'd accept `E: Executor<impl Future<Item=(), Error=()>`, since there isn't really a reason to tell anyone the exact type that is being executed. If I delete from this export, it's essentially doing that. It's a type that implements the future we want, but users can't name it (as would be the case with `impl Future`), and seems best to me.